includes/Modules/Analytics_4/Conversion_Reporting/Conversion_Reporting_Events_Sync.php

```diff
@@ -73,17 +97,44 @@ public function sync_detected_events() {
 			return;
 		}
 
+		// Get current stored detected events.
+		$settings              = $this->settings->get();
+		$saved_detected_events = isset( $settings['detectedEvents'] ) ? $settings['detectedEvents'] : array();
+
 		// phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
 		if ( empty( $report->rowCount ) ) {
 			$this->settings->merge( array( 'detectedEvents' => array() ) );
 
+			if ( ! empty( $this->transients->get( static::DETECTED_EVENTS_TRANSIENT ) ) ) {
+				$this->transients->delete( static::DETECTED_EVENTS_TRANSIENT );
+			}
```

Comment:
This condition is probably redundant. We can call `delete` unconditionally, it will have the same effect no matter we have data or not there:
```suggestion
			$this->transients->delete( static::DETECTED_EVENTS_TRANSIENT );
```